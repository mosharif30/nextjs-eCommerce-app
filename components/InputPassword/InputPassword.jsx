/* eslint-disable react/prop-types */
import React from 'react'

/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react'
import Input from '../Input/Input'

function InputPassword({ placeholder }) {
  return (
    <div
      css={css`
        width: 100%;
        direction: ltr;
      `}
    >
      <Input type="password" placeholder={placeholder} />
    </div>
  )
}

export default InputPassword
