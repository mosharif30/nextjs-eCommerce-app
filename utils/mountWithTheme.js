import { ThemeProvider } from '@emotion/react'
import theme from '../configs/theme'
import renderer from 'react-test-renderer'

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  )
}
export default mountWithTheme
