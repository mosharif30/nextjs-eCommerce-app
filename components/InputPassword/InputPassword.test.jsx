import mountWithTheme from '../../utils/mountWithTheme'
import InputPassword from './InputPassword'
describe('InputPassword', () => {
  test('render correctly', () => {
    const component = mountWithTheme(
      <InputPassword>sample Button</InputPassword>
    )
    const tree = component
    expect(tree).toMatchSnapshot()
  })
  test('render correctly placeholder', () => {
    const component = mountWithTheme(
      <InputPassword placeholder={'sample placeholder'}>
        sample Button
      </InputPassword>
    )
    const tree = component
    expect(tree).toMatchSnapshot()
  })
})
