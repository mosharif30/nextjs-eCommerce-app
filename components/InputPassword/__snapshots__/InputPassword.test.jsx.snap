// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword render correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-c8mvfn-InputPassword{width:100%;direction:ltr;}",
      }
    }
    data-emotion="css c8mvfn-InputPassword"
  />,
  <div
    className="css-c8mvfn-InputPassword"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-i3lbul-Input{width:100%;word-break:break-word;border:2px solid #ecf0f1;background-color:#ecf0f1;margin:0px;outline:2px;font-size:1em;color:#2c3e50;padding:0px;height:32px;}.css-i3lbul-Input:focus{border-bottom:4px solid #1abc9c;background-color:#f5f6fa;}",
        }
      }
      data-emotion="css i3lbul-Input"
    />
    <input
      aria-label="input"
      className="css-i3lbul-Input"
      onChange={[Function]}
      type="password"
    />
  </div>,
]
`;

exports[`InputPassword render correctly placeholder 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-c8mvfn-InputPassword{width:100%;direction:ltr;}",
      }
    }
    data-emotion="css c8mvfn-InputPassword"
  />,
  <div
    className="css-c8mvfn-InputPassword"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-i3lbul-Input{width:100%;word-break:break-word;border:2px solid #ecf0f1;background-color:#ecf0f1;margin:0px;outline:2px;font-size:1em;color:#2c3e50;padding:0px;height:32px;}.css-i3lbul-Input:focus{border-bottom:4px solid #1abc9c;background-color:#f5f6fa;}",
        }
      }
      data-emotion="css i3lbul-Input"
    />
    <input
      aria-label="input"
      className="css-i3lbul-Input"
      onChange={[Function]}
      placeholder="sample placeholder"
      type="password"
    />
  </div>,
]
`;
